### PR TITLE
Feature/reservation more features

### PR DESCRIPTION
**Pull request by 0sugo**
In this PR I  did the following:
 - Ensured a user can only fetch their data.
 - Fixed a bug on the create feature

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,9 +5,11 @@
         registrations: 'api/v1/registrations',
         sessions: 'api/v1/sessions'
       }
+      resources :users do
+        resources :reservations, only: %i[index create destroy]
+      end
       resources :doctors, only: [:index, :show, :new, :create]
       resources :reservations, only: [:index, :create, :destroy]
```
Comment (oguntayomicheal) [inline config/routes.rb]: Well done implementing the routes for reservations 👏 .  I noticed the routes for reservations is declared twice, but as reservations is associated with users, I will suggest removing line 12 as it is not needed.

**Review by oguntayomicheal — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @0sugo :wave:

## Highlights :1st_place_medal:

✔️ Implemented all requirements for reservations
✔️ Descriptive PR Summary
✔️ Following Linters


Good job on implementing all of the requirements! :blush: Kindly check below for my comments.

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

File: config/routes.rb (hunk 1 above)
Comment (0sugo) [reply config/routes.rb]: Thanks brother @oguntayomicheal .Resolved

**Review by oguntayomicheal — APPROVED** [review]
Hello @0sugo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Implemented the required changes ✔✔
- Linters are passing ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**